### PR TITLE
Millix 130

### PR DESCRIPTION
**Pull request by matiasne**

### CODE REVIEW

File: src/js/components/loader.jsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import React from "react";
+import { loaderStatus } from "./utils/loader-status";
+import { Oval } from  'react-loader-spinner'
```
Comment (millix-alex) [inline src/js/components/loader.jsx]: it looks like react-loader-spinner package is not needed anymore. if yes, remove it

File: src/js/components/loader.jsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  return (
+
+    <div className="loader-container">
+      <div className="loader">
+        <div className="circle">         
+        </div>
+      </div>
+    </div>
+
+    /*<div className="loader-container">
```
Comment (millix-alex) [inline src/js/components/loader.jsx]: never push commented code to the repository

File: src/js/components/unlock-wallet-view.jsx [hunk 3]
```diff
@@ -48,6 +48,10 @@ class UnlockWalletView extends Component {
 
     isPrivateKeyExist() {
         API.getIsPrivateKeyExist().then(response => {
+            
+
```
Comment (millix-alex) [inline src/js/components/unlock-wallet-view.jsx]: why did you add 4 empty lines here?

File: yarn.lock [hunk 4]
```diff
@@ -2,12 +2,12 @@
 # yarn lockfile v1
 
 
-"@babel/code-frame@^7.0.0", "@babel/code-frame@^7.10.4", "@babel/code-frame@^7.12.13", "@babel/code-frame@^7.5.5":
-  "integrity" "sha512-HV1Cm0Q3ZrpCR93tkWOYiuYIgLxZXZFVG2VgK+MBWjUqZTundupbfx2aXarXuw5Ko5aMcjtJgbSs4vUGBS5v6g=="
-  "resolved" "https://registry.npmjs.org/@babel/code-frame/-/code-frame-7.12.13.tgz"
-  "version" "7.12.13"
+"@babel/code-frame@^7.0.0", "@babel/code-frame@^7.10.4", "@babel/code-frame@^7.16.7", "@babel/code-frame@^7.5.5":
```
Comment (millix-alex) [inline yarn.lock]: it doesn't look like yarn.lock needs that many changes

File: src/index.js [hunk 5]
```diff
@@ -60,6 +60,8 @@ import API from './js/api';
 import ntp from './js/core/ntp';
 import moment from 'moment';
 import localforage from 'localforage';
+import Loader from "./js/components/loader";
+import { loaderStatus } from "./js/components/utils/loader-status";
```
Comment (millix-alex) [inline src/index.js]: doesn't seem you need loaderStatus here

File: src/js/components/loader.jsx [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+  // Checking if loadingStatus is true the return the LoadingContainer
+  // Else return null
+  render() {
+    const { loadingStatus } = this.state;
+    return loadingStatus ? <LoadingContainer /> : null;
+  }
+}
+
+export default Loader;
```
Comment (millix-alex) [inline src/js/components/loader.jsx]: add empty line at the end of the file

File: src/js/components/utils/loader-status.jsx [hunk 7]
```diff
@@ -0,0 +1,4 @@
+import { Subject } from "rxjs";
+
+// laderStatus as Subject, It will work as both observer and obserable
+export const loaderStatus = new Subject();
```
Comment (millix-alex) [inline src/js/components/utils/loader-status.jsx]: add empty line at the end of the file

File: package.json [hunk 8]
```diff
@@ -23,14 +23,16 @@
     "react-bootstrap": "^2.1.0",
     "react-bootstrap-typeahead": "^5.2.0",
     "react-dom": "^17.0.2",
+    "react-error-overlay": "6.0.9",
+    "react-loader-spinner": "^6.0.0-0",
```
Comment (millix-alex) [inline package.json]: remove react-loader-spinner from packages

**Comment by matiasne** [comment]
Corrections made
